openapi.yaml

```diff
@@ -0,0 +1,712 @@
+          description: ID of the service
+          required: true
+          schema:
+            type: string
+      responses:
+        '200':
+          description: Successful operation _(service data)_
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/Service'
+        '401':
+          description: Unauthorized
+        '403':
+          description: Forbidden
+        '404':
+          description: Not found
+        '429':
+          description: Too many requests
+      security:
+        - subscription_key: []
+    put:
+      tags:
+        - services
+      summary: Update service
+      description: |
+        Update an existing service by by `serviceId`. <br><br>_Note: This API operation needs the __Manage API key__ owned by the service user, otherwise `403 forbidden` will be returned to the caller._
+      operationId: updateService
+      parameters:
+        - name: serviceId
+          in: path
+          description: ID of the service
+          required: true
+          schema:
+            type: string
+      requestBody:
+        description: Updated service payload
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/ServicePayload'
+        required: true
+      responses:
+        '200':
+          description: Successful operation _(service updated)_
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/Service'
+        '400':
+          description: Invalid payload
+        '401':
+          description: Unauthorized
+        '403':
+          description: Forbidden
+        '404':
+          description: Not found
+        '429':
+          description: Too many requests
+        '500':
+          description: Internal server error _(service cannot be updated)_
+      security:
+        - subscription_key: []
+    delete:
+      tags:
+        - services
+      summary: Delete service
+      description: |
+        Delete a service by `serviceId`. <br>This operation is a logical deletion: the service persist in user services, but it's status become
+        `deleted`<br>_Note: This API operation needs the __Manage
+        API key__ owned by the service user, otherwise `403 forbidden` will be returned to the caller._
+      operationId: deleteService
+      parameters:
+        - name: serviceId
+          in: path
+          description: ID of the service
+          required: true
+          schema:
+            type: string
+      responses:
+        '200':
+          description: Successful operation _(service deleted)_
+        '401':
+          description: Unauthorized
+        '403':
+          description: Forbidden
+        '404':
+          description: Not found
+        '429':
+          description: Too many requests
+        '500':
+          description: Internal server error _(service cannot be deleted)_
```

Comment:
Right. Generally speaking we should have:
* Operations that cannot be completed because of data (provided or persisted) --> 4xx
* Operations that failed for technical issues --> 5xx

The meaning for the user is, respectively, "I'm doing something wrong, let's find out" and "The system is having trouble, let's try again in a while".